### PR TITLE
refactor isEnabled in separate package configcheck

### PR DESCRIPTION
**Pull request by mackjmr**
### What does this PR do?

This PR refactors out logic from `github.com/DataDog/datadog-agent/comp/otelcol/otlp` into `github.com/DataDog/datadog-agent/comp/otelcol/otlp/configcheck`. This allows switching import from `github.com/DataDog/datadog-agent/comp/otelcol/otlp` to `github.com/DataDog/datadog-agent/comp/otelcol/otlp/configcheck` in other agent pkgs, significantly reducing the number of otel deps in some agent binaries (see golang_deps_diff)

### Describe how to test/QA your changes
N/A. Refactor no functional change.

### CODE REVIEW

File: comp/otelcol/otlp/configcheck/configcheck.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+	return hasSection(cfg, coreconfig.OTLPLogsEnabled) && cfg.GetBool(coreconfig.OTLPLogsEnabled)
+}
+
+func hasSection(cfg config.Reader, section string) bool {
+	// HACK: We want to mark as enabled if the section is present, even if empty, so that we get errors
+	// from unmarshaling/validation done by the Collector code.
+	//
+	// IsSet won't work here: it will return false if the section is present but empty.
+	// To work around this, we check if the receiver key is present in the string map, which does the 'correct' thing.
+	_, ok := ReadConfigSection(cfg, coreconfig.OTLPSection).ToStringMap()[section]
```
Comment (GustavoCaso) [inline comp/otelcol/otlp/configcheck/configcheck.go:77]: maybe something to consider for a follow up PR, but it seems that when calling `IsEnabled ` and `HasLogsSectionEnabled ` we parse the `coreconfig.OTLPSection`. Looking at the code, it seems that we are doing a bit of copying and creation of `confmap`

Maybe we couild hoist the result from calling `ReadConfigSection(cfg, coreconfig.OTLPSection)` using a `sync.Once` and avoid parsing the same section multiple times 😄

**Review by GustavoCaso — APPROVED** [review]
Nice work 🎉

**Review by ichinaski — APPROVED** [review]

**Review by duncanista — APPROVED** [review]

File: comp/otelcol/otlp/configcheck/configcheck.go (hunk 1 above)
Comment (mackjmr) [reply comp/otelcol/otlp/configcheck/configcheck.go:77]: Good point, created a task to look into improving this.

**Review by IbraheemA — APPROVED** [review]

**Pull request merged**